### PR TITLE
{lib}[foss/2019b] carma v2.01

### PR DESCRIPTION
**Pull request by kErica**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by robert-mijakovic** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+toolchain = {'name': 'foss', 'version': '2019b'}
+
+source_urls = ['http://utopia.duth.gr/~glykos/progs/']
+sources = ['%(name)s.tar.gz']
+checksums = ['f32cfc2e348691f252d3593e281303d1202e31befba14565aba022331d50b9f3']
+
+preinstall_cmd = 'cd %(builddir)s/%(name)s-%(version)s/src && '
+preinstall_cmd += '$CC $CFLAGS -O -c Color_carma.c && '
+preinstall_cmd += '$FC $FFLAGS $LIBBLAS -O Color_carma.o -o carma &&'
+preinstall_cmd += 'cp -p carma ../bin/'
```
Comment (lexming) [inline easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb:25]: In this case I recommend to use the `CmdCp` easyblock, which provides some more structure to perform the actions in the `preinstall_cmd`:
```suggestion
start_dir = 'src'

cmds_map = [('.*', "$CC $CFLAGS -O -c Color_carma.c && $FC $FFLAGS $LIBBLAS -O Color_carma.o -o carma")]
files_to_copy = [(['carma'], 'bin')]
```

File: easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# #
+# This is a contribution from HPCNow! (http://hpcnow.com)
+# Copyright::   HPCNow!
+# Authors::     Helena Gomez <helena.gomez@hpcnow.com>, Erica Bianco, Danilo Gonzalez, Jordi Blasco
+# License::     GPL-v3.0
+# #
+
+easyblock = 'Tarball'
+
+name = 'carma'
```
Comment (lexming) [inline easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb:10]: According to the homepage, the name should be capitalized:
```suggestion
name = 'Carma'
```

File: easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+preinstall_cmd += '$CC $CFLAGS -O -c Color_carma.c && '
+preinstall_cmd += '$FC $FFLAGS $LIBBLAS -O Color_carma.o -o carma &&'
+preinstall_cmd += 'cp -p carma ../bin/'
+
+sanity_check_paths = {
+    'files': ['bin/carma'],
+    'dirs': [''],
+}
+
+tests = ['%(installdir)s/bin/carma']
```
Comment (lexming) [inline easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb:32]: I believe that you want to run `sanity_check_commands` here:
```suggestion
sanity_check_commands = ['carma']
```

File: easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+moduleclass = 'lib'
```
Comment (lexming) [inline easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb:34]: Software to execute molecular dynamics are usually classified in `chem`
```suggestion
moduleclass = 'chem'
```

**Review by lexming — CHANGES_REQUESTED** [review]
Thanks for your contribution @kErica . I recommend using the `CmdCp` easyblock instead of `Tarball` with this software. Please see my code suggestions below

File: easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb (hunk 4 above)
Comment (kErica) [reply easybuild/easyconfigs/c/carma/carma-2.01-foss-2019b.eb:34]: All suggestions applied, thanks @lexming

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node313.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/3087ede0ebdf6250ce72e3bb05f102dd for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
Going in, thanks @kErica !

**Pull request merged**